docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc

```diff
@@ -0,0 +1,131 @@
+====
+[start=4]
+. Click **Import code**. 
+
+[IMPORTANT]
+====
+If you want to use a custom pipeline build, or to grant access to and install the {ProductName} GitHub app on the repository you indicated, you must have the necessary GitHub permissions. If you don't own that repository, or if you don't have the necessary GitHub permissions, you must fork that repository and then specify the new fork URL to add a new component. 
+====
+
+////
+[TIP]
+====
+If {ProductName} can’t access your Git repository, see <<Granting App Studio access to your Git repository>> for help.
+====
+////
+
+{ProductName} scans and analyzes your Git repository to determine how many components to add to your application and which runtime to use. The types of files in your repository, and also your Git directory file hierarchy, determine the number and type of components to add. For example, {ProductName} looks for devfiles and Dockerfiles, and also language-specific build configuration files like pom.xml for Java or package.json for Node.js.
+
+[NOTE]
+====    
+{ProductName} can build applications that are written in the following language stacks and frameworks:
+
+    [cols=2*,options="header"]
+|===
+|Language
+|Framework
+
+|Java
+|Spring Boot, Quarkus
+
+|JavaScript
+|Node.js, Express
+
+|Python
+|Flask
+
+|Go
+|N/A
+|===
+====
+
+== Using a code sample
+
+. Optional: In the code sample **Search** field, begin typing the name of the code you want to work with. If {ProductName} has code samples in that language stack, the code sample results filter accordingly.
+. Optional: If you see a language stack or framework that you want to work with, click either its **Open Git repository** or **Import sample** link.    
+* If you click **Open Git repository**, a new tab opens to the GitHub repository for the code framework you chose.
+* If you click **Import sample**, {ProductName} creates and deploys a new application with the sample code you chose. Check out the open **Overview** tab to monitor your continuous integration/continuous deployment (CI/CD) activity.
+
+.What's next
+**Configure your components for deployment** opens. From here, configure your components before you deploy them. You can also review or change any values that {ProductName} populated for you. When you're finished here, you're one click away from creating your application.
+
+//// 
+From JSese: May 15, 2023, Commenting out per Matt Reid and Christian V. Support for private repos is post-Summit.
+[discrete]
+== Granting App Studio access to your Git repository
+
+If {ProductName} can’t access your Git repository, the following message displays: “We can’t access your repository, probably because it’s either private or the URL is incorrect.” If your repository is private, choose one of the following two ways to allow {ProductName} to interact with your repository:
```

Comment:
Does it make sense to explicitly mention somewhere that the Git repository URL should be referencing a public GitHub repository? Until the summit that is. WDYT?

Reply:
thanks, @xbaran4! i added that detail. 

do you recommend a way to manage all these little things for after Summit?

Reply:
@jseseCCS Something big like private repo support will be tied to a feature (RHTAP-304 in this case), so you could add a doc epic there to come back and make sure everything is accurate. The dockerfile field question above isn't scoped yet, but we can talk to Missy and the front end devs about ensuring Docs is kept in the loop on any smaller import experience changes.

Reply:
sounds great, @mattreid thank you!